crates/biome_deserialize/src/impls.rs

```diff
@@ -0,0 +1,571 @@
+
+/// A String representation of a number (integer, float).
+/// The format should be parsable by Rust numeric types.
+#[derive(Eq, PartialEq, Clone)]
+pub struct TokenNumber(pub(crate) TokenText);
+impl TokenNumber {
+    pub fn text(&self) -> &str {
+        self.0.text()
+    }
+}
+impl Deserializable for TokenNumber {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        struct Visitor;
+        impl DeserializationVisitor for Visitor {
+            type Output = TokenNumber;
+            const EXPECTED_TYPE: ExpectedType = ExpectedType::NUMBER;
+            fn visit_number(
+                self,
+                value: TokenNumber,
+                _range: TextRange,
+                _diagnostics: &mut Vec<DeserializationDiagnostic>,
+            ) -> Option<Self::Output> {
+                Some(value)
+            }
+        }
+        value.deserialize(Visitor, diagnostics)
+    }
+}
+
+impl Deserializable for () {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        diagnostics.push(DeserializationDiagnostic::new_incorrect_type(
+            ExpectedType::empty(),
+            value.range(),
+        ));
+        None
+    }
+}
+
+impl Deserializable for bool {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        struct Visitor;
+        impl DeserializationVisitor for Visitor {
+            type Output = bool;
+            const EXPECTED_TYPE: ExpectedType = ExpectedType::BOOL;
+            fn visit_bool(
+                self,
+                value: bool,
+                _range: TextRange,
+                _diagnostics: &mut Vec<DeserializationDiagnostic>,
+            ) -> Option<Self::Output> {
+                Some(value)
+            }
+        }
+        value.deserialize(Visitor, diagnostics)
+    }
+}
+
+impl Deserializable for f32 {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        let range = value.range();
+        let value = TokenNumber::deserialize(value, diagnostics)?;
+        if let Ok(value) = value.text().parse::<Self>() {
+            return Some(value);
+        }
+        let diagnostic =
+            DeserializationDiagnostic::new("The number should be a float representable on 32 bits")
+                .with_range(range);
+        diagnostics.push(diagnostic);
+        None
+    }
+}
+
+impl Deserializable for f64 {
+    fn deserialize(
+        value: impl DeserializableValue,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        let range = value.range();
+        let value = TokenNumber::deserialize(value, diagnostics)?;
```

Comment:
When I read the documentation of `TokenNumber`, I thought it was meant to deserialize cases like `"49"`, although here we are using it for a `f64`. So, it's possible I don't understand this implementation or the documentation of `TokenNumber` is incorrect.